sphinx/domains/c/_symbol.py

```diff
@@ -622,12 +574,11 @@ def direct_lookup(self, key: LookupKey) -> Symbol | None:
             Symbol.debug_print("direct_lookup:")
             Symbol.debug_indent += 1
         s = self
-        for name, id_ in key.data:
+        for ident, id_ in key.data:
             res = None
-            for cand in s._children:
-                if cand.ident == name:
-                    res = cand
-                    break
+            name = str(ident)
+            if name in s._childrenByName:
+                res = s._childrenByName[name]
```

Comment:
```suggestion
			res = s._childrenByName.get(str(ident))
```

and remove the `res = None` above (I cannot suggest on non-modified lines).

Reply:
Good suggestion. I will also simplify the method a bit more after this change.